### ISSUE TITLE
CycleGAN produces identical images from input

### ISSUE THREAD

**Issue by NeilDG** [junyanz/pytorch-CycleGAN-and-pix2pix#1080]
I'm experimenting with some style transfer using CycleGAN. However, I encountered a major problem:

Let A = CCTV image of road traffic with some level of noise and haze.
Let B = GTA 5 image dataset from (http://efrosgans.eecs.berkeley.edu/cyclegta/gta.zip)

My goal is to have the style of B transferred to A. CycleGAN starts to produce identical images like A (MSE loss becomes very small between real and fake samples), indicating that no style transfer was performed at all. 

My hypotheses:
- I have added identity loss as stated in the paper but I think CycleGAN prioritizes identity loss too much. Identity loss set to 1.0.
- Features of B is not easily transferrable to A and vice versa.
- Different camera angle between A and B is not supported by CycleGAN training.
- Noisy images from A greatly affects training.

Would like to hear any thoughts and suggestions for this! Thank you very much!

**Comment by junyanz** [comment]
- Reducing the identity loss might help. 
- It works better when the camera angle is similar across domains. 
- It works better if you train the model on cropped patches rather than entire image. See #586 for more details and hyper-parameters.

**Comment by NeilDG** [comment]
Thanks for the insights. Will try bullet #3.

**Comment by ibro45** [comment]
@NeilDG how did it go? I encountered the same problem. Thanks!

**Comment by NeilDG** [comment]
> @NeilDG how did it go? I encountered the same problem. Thanks!

I realized my image domain and methodology is not suitable for the vanilla CycleGAN. I've discontinued my experiments regarding this. But so far I did the following that somewhat fixed the solution (but produced blurry images):

- Check if you're correctly passing the images in the cycle-consistency loss. See their paper. I have mistakenly mixed images A and B during training. This might be the primary issue.
- Identity loss only changes the color of my images, but not necessarily enhancing them, which is what I want. Reducing this created blurry images.
- Training on randomly cropped patches as suggested above greatly helps rather than training them as a whole. I've started using this approach on my future GANs.
